registrations/permissions.py

```diff
@@ -66,4 +93,57 @@ def has_object_permission(
         if request.method == "DELETE":
             return obj.can_be_deleted_by(request.user)
 
+        if request.method in ("PUT", "PATCH"):
+            return self._has_object_update_permission(request, obj)
+
+        return obj.can_be_edited_by(request.user)
+
+
+class CanAccessSignupGroup(SignUpBasePermission):
+    @staticmethod
+    def _has_object_update_permission(request: Request, obj: SignUpGroup) -> bool:
+        if request.user.is_superuser or request.user.is_registration_admin_of(
+            obj.publisher
+        ):
+            return True
+
+        def get_signups_keys():
+            keys = set()
+            for signup in request.data.get("signups", []):
+                keys |= set(signup.keys())
+            return keys
+
+        if obj.registration.registration_user_accesses.filter(
+            email=request.user.email
+        ).exists():
+            # Registration user can only change presence_status and nothing else,
+            # and they need to be strongly identified.
+            data_keys = get_signups_keys()
+            return request.user.is_strongly_identified and (
+                data_keys == {"id", "registration", "presence_status"}
+                or data_keys == {"id", "presence_status"}
+            )
+        elif obj.created_by_id == request.user.id:
+            # User who created the signup, and is not one of the signup admins,
+            # is allowed to change all other data except presence_status.
+            data_keys = get_signups_keys()
+            return "presence_status" not in data_keys
+
+        return False
+
+    def has_object_permission(
+        self, request: Request, view: APIView, obj: SignUpGroup
+    ) -> bool:
+        if isinstance(request.user, ApiKeyUser):
+            user_data_source, _ = view.user_data_source_and_organization
+            # allow only if the api key matches instance data source
+            if obj.data_source != user_data_source:
+                return False
+
+        if request.method == "DELETE":
+            return obj.can_be_deleted_by(request.user)
+
+        if request.method in ("PUT", "PATCH") and request.data.get("signups"):
+            return self._has_object_update_permission(request, obj)
+
```

Comment:
`CanAccessSignupGroup` could inherit `CanAccessSignup`. In that case you could only override `_has_object_update_permission`

has_object_permissions is duplicated, if I see it correctly.